src/globals.c

```diff
@@ -311,20 +311,23 @@ bool set_path_prefix(char *path)
 		return false;
 	}
 
-	ret = stat(path, &statbuf);
-	// If path doesn't exit tries to create it
-	if (ret < 0 && errno == ENOENT) {
+	new_path = realpath(path, NULL);
+	if (!new_path) {
 		if (mkdir_p(path) < 0) {
 			goto error;
 		}
-		ret = stat(path, &statbuf);
+		new_path = realpath(path, NULL);
 	}
-	if (ret < 0 || !S_ISDIR(statbuf.st_mode)) {
+	if (!new_path) {
 		goto error;
 	}
 
-	new_path = realpath(path, NULL);
-	if (!new_path) {
+	ret = stat(new_path, &statbuf);
+	// If path doesn't exit tries to create it
```

Comment:
this comment is now in the wrong section of the code

Reply:
True. I'll move it